t/subname.t

```diff
@@ -55,8 +55,7 @@ sub caller3_ok {
     # this is apparently how things worked before 5.16
     utf8::encode($expected) if $] < 5.016 and $ord > 255;
     # before 5.16 and after v5.25.2c names with NUL are stripped
-    my $strip_nul = ($] < 5.016 or ($Config{usecperl} and $] >= 5.025002));
-    if (!$ord and $strip_nul) {
+    if (!$ord and ($] < 5.016 or ($Config{usecperl} and $] >= 5.025002))) {
```

Comment:
These changes appear out of sync with the Sub-Name distribution (the corresponding file there is t/exotic_names.t).  It would be appreciated if the changes go there first, with suitable documentation that we understand what the problem(s) are.